.azure-pipelines/ultimate-pipeline.yml

```
@@ -898,6 +945,140 @@ stages:
     - script: tracer\build.cmd CheckBuildLogsForErrors
       displayName: Check logs for errors
 
+- stage: exploration_tests_linux
+  condition: and(succeeded(), eq(variables['isScheduledBuild'], 'False'), or(eq(stageDependencies.check_changes.outputs['check_changes_job.check_changes_step.IsProfilerChanged'],'True'), eq(stageDependencies.check_changes.outputs['check_changes_job.check_changes_step.IsDebuggerChanged'], 'True')))
```

nit: multiline for readability
```suggestion
  condition: >
    and(
      succeeded(),
      eq(variables['isScheduledBuild'], 'False'),
      or(
        eq(stageDependencies.check_changes.outputs['check_changes_job.check_changes_step.IsProfilerChanged'], 'True'),
        eq(stageDependencies.check_changes.outputs['check_changes_job.check_changes_step.IsDebuggerChanged'], 'True')
      )
    )
```